src/main/webapp/app/exercises/programming/manage/grading/programming-exercise-configure-grading.component.ts

```diff
@@ -589,11 +597,31 @@ export class ProgrammingExerciseConfigureGradingComponent implements OnInit, OnD
     /**
      * Executes filtering on all available test cases with the specified params.
      */
-    updateTestCaseFilter = () => {
+    updateTestCaseFilter() {
         this.filteredTestCasesForTable = !this.showInactiveValue && this.testCases ? this.testCases.filter(({ active }) => active) : this.testCases;
         this.filteredTestCasesForCharts = this.filteredTestCasesForTable;
         this.backupTestCases = this.filteredTestCasesForTable;
-    };
+    }
+
+    /**
+     * Calculates the rounded points awarded for passing each test
+     */
+    updateTestPoints() {
+        this.testCasePoints = {};
+
+        if (!this.testCases) {
+            return;
+        }
+
+        const totalWeight = this.testCases.reduce((sum, testCase) => sum + testCase.weight!, 0);
+        const maxPoints = this.programmingExercise.maxPoints!;
+        this.testCases.forEach((testCase) => {
+            if (testCase.testName) {
```

Comment:
Why do we have this distinction? What is the use case here? Do we hide test cases without a name? I don't see an indication of that.